Frontend/ui-customer/src/Components/InputView/View.purs

```diff
@@ -0,0 +1,252 @@
+view push state = 
+  linearLayout
+    [ height WRAP_CONTENT
+    , width MATCH_PARENT
+    , orientation VERTICAL
+    , padding $ PaddingHorizontal 16 16
+    , background Color.black900 
+    ]
+    [ linearLayout
+      [ height WRAP_CONTENT
+      , width MATCH_PARENT
+      , orientation HORIZONTAL
+      ]  
+      [ backPressView state push
+      , dateTimePickerButton state push
+      ]
+    , linearLayout
+      [ height WRAP_CONTENT
+      , width MATCH_PARENT
+      , padding $ PaddingVertical 16 16
+      , gravity CENTER_VERTICAL
+      ][  inputImageView push state
+        , textViews push state
+      ]
+    ]
+
+backPressView :: forall w. InputViewConfig -> (Action -> Effect Unit) -> PrestoDOM (Effect Unit) w
+backPressView config push = 
+  linearLayout
+  [ height MATCH_PARENT
+  , width WRAP_CONTENT
+  , padding $ config.backIcon.padding 
+  , onClick push $ const BackPressed
+  ][  imageView
+      [ height $ config.backIcon.height 
+      , width $ config.backIcon.width
+      , imageWithFallback config.backIcon.imageName
+      ]
+    , textView $
+      [ text $ config.headerText
+      , visibility $ boolToVisibility $ config.headerVisibility
+      , color Color.white900  
+      , margin $ MarginLeft 8
+      ] <> (FontStyle.subHeading2 LanguageStyle)
+  ]
+
+dateTimePickerButton :: forall w. InputViewConfig -> (Action -> Effect Unit) -> PrestoDOM (Effect Unit) w
+dateTimePickerButton config push = 
+  linearLayout
+  [ height MATCH_PARENT
+  , width MATCH_PARENT
+  , orientation HORIZONTAL 
+  , margin $ MarginTop 16
+  , visibility config.suffixButton.visibility
+  ]
+  [ linearLayout
+    [ height WRAP_CONTENT
+    , weight 1.0
+    ][]
+  , linearLayout
+    [ height WRAP_CONTENT
+    , width WRAP_CONTENT
+    , background "#303440"
+    , cornerRadius 4.0
+    , padding $ config.suffixButton.padding
+    , gravity config.suffixButton.gravity
+    , onClick push $ const DateTimePickerButtonClicked
+    ]
+    [ imageView
+      [ imageWithFallback $ fetchImage FF_ASSET config.suffixButton.prefixImage
+      , height $ V 16
+      , width $ V 16
+      , margin $ MarginTop 1
+      ]
+    , textView $
+      [ text config.suffixButton.text
+      , margin $ MarginHorizontal 4 4
+      , color Color.black600
+      ] <> FontStyle.subHeading2 LanguageStyle
+    , imageView
+      [ imageWithFallback $ fetchImage FF_ASSET config.suffixButton.suffixImage
+      , height $ V 16
+      , width $ V 16
+      , margin $ MarginTop 1
+      ]
+    ]
+  ]
+
+
+inputImageView :: forall w. (Action -> Effect Unit ) -> InputViewConfig -> PrestoDOM (Effect Unit) w
+inputImageView push config = let 
+  lastIndex = (length config.inputView) - 1
+  in
+  linearLayout
+    [ height WRAP_CONTENT
+    , width $ config.imageLayoutWidth 
+    , orientation VERTICAL
+    , margin $ config.imageLayoutMargin
+    , gravity CENTER
+    , visibility config.imageLayoutVisibility
+    ]
+    ( mapWithIndex 
```

Comment:
put the whole layout inside a function instead of just the imageView (prefixImageView)